restapi/src/main/java/io/stargate/web/resources/RowResource.java

```diff
@@ -400,12 +400,15 @@ private static Object filterToValue(Object val, String column, Table tableData)
   private List<Where<?>> buildWhereClause(PathSegment id, Table tableMetadata) {
     List<String> values = idFromPath(id);
 
-    final List<Column> keys = tableMetadata.partitionKeyColumns();
-    if (keys.size() != values.size()) {
-      throw new IllegalArgumentException("not enough partition keys provided");
+    final List<Column> keys = tableMetadata.primaryKeyColumns();
+    if (keys.size() < values.size()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Invalid number of key values required (%s). All partition key columns values are required plus 0..all clustering columns values in proper order.",
```

Comment:
Do we actually validate the order of CC columns or will persistence do that?

Reply:
No, there really isn't a good way to do validation since we only have the values but not the keys/column names.